### PR TITLE
feature: allow custom token representations

### LINKED ISSUE #481: feature: Custom Implementations for Bridge Token
- goal: let teams manage the implementation of custom tokens

features: 
- governance can set a new address for a specific token 
- migrate from system bridge token to custom bridge token (burn system bridge token, mint custom bridge token)

### PR DESCRIPTION
**Pull request by prestwich**
closes #481 

@anna-carroll please review the design

### CODE REVIEW

**Comment by prestwich** [comment]
this needs tests

**Review by anna-carroll — COMMENTED** [review]
looking great to me 🤠 couple small comments.

we'll need to rebase off current main, too

File: solidity/optics-xapps/contracts/bridge/BridgeRouter.sol [hunk 1]
```diff
@@ -192,4 +192,57 @@ contract BridgeRouter is Router, TokenRegistry {
...
+        external
+        onlyOwner
+    {
+        require(_id.length == 36, "!tokenId");
+        bytes29 _tokenId = _id.ref(0);
+        bytes32 _idHash = _tokenId.keccak();
+
+        representationToCanonical[_custom].domain = _tokenId.domain();
+        representationToCanonical[_custom].id = _tokenId.id();
+        canonicalToRepresentation[_idHash] = _custom;
```
[the remark this answers is not in the document]
Comment (prestwich) [reply solidity/optics-xapps/contracts/bridge/BridgeRouter.sol:235]: sanity checks are good. gonna add a mint/burn cycle here

**Review by anna-carroll — APPROVED** [review]

**Comment by prestwich** [comment]
fast liquidity test bytecode is now too large to deploy. need to break it  up 😬

**Review by anna-carroll — APPROVED** [review]

**Pull request merged**